### PR TITLE
Add interactive graphs generation to npf-run (npf-web-extension)

### PR DESCRIPTION
**Pull request by dpcodebiz**
# Description
This PR adds a new argument to `npf-run` (`--web [export_path]`). This new argument allows users to export data to html format via npf-web-extension (https://github.com/dpcodebiz/npf-web-extension). This parameter is optional and can be used along with all other parameters (`--single-output`, etc.)

# Motivation
Npf-run is only able to generate static graphs and this extension goes further by allowing the user to export an instance-based interface that runs in any browser and provides data visualization capabilities.

# Relevant changes
- added handlers for `--web [export_path]` argument
- added `npf-web-extension` to dependencies

# Tests
- User testing

### CODE REVIEW

File: npf/grapher.py [hunk 1]
```diff
@@ -184,7 +186,7 @@ def statics(self):
     def dyns(self):
         return [var for var,values in self.vars_values.items() if len(values) > 1]
 
-    #Convert the series into te XYEB format (see types.dataset)
```
Comment (tbarbette) [inline npf/grapher.py:187]: ❤️

File: npf/types/web/web.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+from npf_web_extension import app
+
+def prepare_web_export(datasets, all_results_df, path):
+
+    # Getting parameters and measurements
+    name = "undefined"
+    parameters = []
+    measurements = []
+    for testie, _, runs in datasets:
```
Comment (tbarbette) [inline npf/types/web/web.py:11]: For the same reasons than https://github.com/kohler/click/pull/468 where the name "testie" is inspired from, could you change references of testie to "test"? or npf test if appropriate.

File: npf/types/web/web.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      name = testie.get_title()
+      for run, results in runs.items():
+        measurements = list(results.keys())
+        parameters = list(run.variables.keys())
+        break
+      break
+
+    # Preparing configuration data
+    configurationData = {
```
Comment (tbarbette) [inline npf/types/web/web.py:20]: Somehow the variable names from test.config["var_names"] should be used to have the right axis names

**Review by tbarbette — CHANGES_REQUESTED** [review]
Almost good:
- See the inline comments. About passing the variable names and use them for axis, it can come later but would be really needed.
- Add one or two unit tests in integration/*.py to verify the code works, and maybe one integration test in integration/integration.sh to check a website is built, but not (yet?) verifying it looks good.

**Comment by tbarbette** [comment]
@dpcodebiz any new on this?

File: npf/types/web/web.py (hunk 3 above)
Comment (dpcodebiz) [reply npf/types/web/web.py:20]: Well, I would do that but it seems that var_names are hardcoded by default (see https://github.com/dpcodebiz/npf/blob/c7e6dc6374d775875027e0d1f7244b515ec05f25/npf/section.py#L690 ). Hence, if I use it, for tests like examples/math.npf, graphs axis would be "LATENCY" which has no sense imo. Thus, should we rather remove the default value first then proceed or just leave it the way it is?

**Comment by dpcodebiz** [comment]
I don't think a unit test is necessary as the small portion of code I added to npf is already being asserted inside the integration test. In other words, https://github.com/dpcodebiz/npf/blob/36ccb1fe9c79f1b0a58c730bb099e08b9a131f12/npf/types/web/web.py#L5-L29 this is the only relevant part worth testing and checking that a .html file is being generated asserts that the method is working properly. What do you think?

File: npf/types/web/web.py (hunk 3 above)
Comment (tbarbette) [reply npf/types/web/web.py:20]: No that's not how it works. The idea for "RESULT-NAME"  is to look for result in the "var_names" dictionnary. Some default names like Latency are there, but the test file can use var_names+={LOG:Logarithm} for math.npf for instance.

**Comment by tbarbette** [comment]
Ok for the tests :) For the variable names I guess it's easy enough.

File: npf/types/web/web.py (hunk 3 above)
Comment (dpcodebiz) [reply npf/types/web/web.py:26]: Okay, understood. I just added it, it should be good now!

**Review by tbarbette — APPROVED** [review]

**Pull request merged**